crates/gen-host-js/src/lib.rs

```diff
@@ -713,12 +734,11 @@ impl Instantiator<'_> {
         func: &Function,
         abi: AbiVariant,
     ) {
-        // Technically it wouldn't be the hardest thing in the world to support
-        // other string encodings, but for now the code generator was originally
-        // written to support utf-8 so let's just leave it at that for now. In
-        // the future when it's easier to produce components with non-utf-8 this
-        // can be plumbed through to string lifting/lowering below.
-        assert_eq!(opts.string_encoding, StringEncoding::Utf8);
+        // Only Utf8 and Utf16 supported for now
+        assert!(matches!(
+            opts.string_encoding,
```

Comment:
Could this option get plumbed through into the `FunctionBindgen` which would push this assert into the `String{Lift,Lower}` portions?